### PR TITLE
Feature to introduce Dev Containers

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*
Introducing  Dev Containers Feature.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by elamaran11** [comment]
@shapirov103  added more tools as you requested. Feel free to merge.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "image": "mcr.microsoft.com/vscode/devcontainers/base:ubuntu",
+    "features": {
+      "ghcr.io/devcontainers/features/node:1": {},
```
Comment (shapirov103) [inline .devcontainer/devcontainer.json:4]: I am getting node 22 that is not the right version for cdk. Getting this large orange warning when running `cdk list`.

```
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!                                                                                                                      !!
!!  This software has not been tested with node v22.1.0.                                                                !!
!!  Should you encounter odd runtime issues, please try using one of the supported release before filing a bug report.  !!
!!                                                                                                                      !!
!!  This software is currently running on node v22.1.0.                                                                 !!
!!  As of the current release of this software, supported node releases are:                                            !!
!!  - ^20.0.0 (Planned end-of-life: 2026-04-30)                                                                         !!
!!  - ^18.0.0 (Planned end-of-life: 2025-04-30)                                                                         !!
!!                                                                                                                      !!
!!  This warning can be silenced by setting the JSII_SILENCE_WARNING_UNTESTED_NODE_VERSION environment variable.        !!
!!                                                                                                                      !!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
let's set node to the supported version (20.x)

File: .devcontainer/postCreateCommand.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# setup autocomplete for kubectl and alias k
+mkdir $HOME/.kube
+chsh -s $(which zsh)
+echo "source <(kubectl completion bash)" >> $HOME/.zshrc
+echo "alias k=kubectl" >> $HOME/.zshrc
+echo "complete -F __start_kubectl k" >> $HOME/.zshrc
+
+# Add Tools for Syntax Highlighting, Auto Suggestions, command shortcurts with oh-my-zsh for AWS, Kubectl, Node, Git.
+sh -c "$(curl -fsSL https://raw.github.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
```
Comment (shapirov103) [inline .devcontainer/postCreateCommand.sh:25]: Missing CDK command. Try running cdk list or anything like. 

lets set an alias cdk="npx cdk"

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (shapirov103) [inline .devcontainer/devcontainer.json:1]: on reset container getting the same error "workspace does not exist, rerun as root"

**Review by shapirov103 — COMMENTED** [review]
@elamaran11 i had to drop my dev container and recreate. 
I am getting the workspace creation issue "try running as root".

In the log file I see the following error: .devcontainer/postCreateCommand.shL permission denied

I suspect my other issue are derivatives of this

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (elamaran11) [reply .devcontainer/devcontainer.json]: Done

File: .devcontainer/postCreateCommand.sh (hunk 2 above)
Comment (elamaran11) [reply .devcontainer/postCreateCommand.sh:26]: Done. But ideally you should not be pre appending `npm install` in `devContainers` and its an anti pattern. Please check this issue - https://stackoverflow.com/questions/55976358/should-a-vscode-dev-container-run-npm-install-as-part-of-its-setup

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (elamaran11) [reply .devcontainer/devcontainer.json:1]: Weird, never faced this.

**Comment by elamaran11** [comment]
> @elamaran11 i had to drop my dev container and recreate. I am getting the workspace creation issue "try running as root".
> 
> In the log file I see the following error: .devcontainer/postCreateCommand.shL permission denied.
> 
> I suspect my other issue are derivatives of this.
> 
> Another question: is there a way to connect to the same machine in the cloud from a different laptop?

@shapirov103 Firstly how are you running `DevPod` are you running from repo branch or pulling the code locally to point th folder. I dont recommend second approach if you have `cdk.out` and `node modules` which makes the `DevPod` heavy and you get these issues. I recommend running it directlly referencing GitHub repo. I never had these issues. Try to recreate from start, we can screen share if you need.

**Comment by shapirov103** [comment]
@elamaran11 please touch something on the PR to retrigger the markdown check that is failing. I don't have an option to restart this GH action.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**